modules/runtime-common/src/main/scala/com/snowplowanalytics/snowplow/runtime/Webhook.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2014-present Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package com.snowplowanalytics.snowplow.runtime
+
+import cats.effect.{Async, Sync}
+import cats.implicits._
+import cats.Show
+import io.circe.{Decoder, Json}
+import io.circe.generic.semiauto._
+import io.circe.syntax._
+import org.http4s.circe.jsonEncoder
+import org.http4s.client.Client
+import org.http4s.{Method, Request}
+import org.http4s.{ParseFailure, Uri}
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import com.snowplowanalytics.iglu.core.circe.implicits._
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaVer, SelfDescribingData}
+import com.snowplowanalytics.snowplow.runtime.AppInfo
+
+trait Webhook[F[_], Alert] {
+  def alert(message: Alert): F[Unit]
+}
```

Comment:
Here I put `Alert` as a type parameter on the trait.  I considered an alternative implementation where it's a type parameter on  the method, like:

```scala
trait Webhook[F[_]] {
  def alert[Alert: Show](message: Alert): F[Unit]
}
```

But I chose against that, because it means in the application code you could call `webhook.alert(something)` with _anything_ that has a show.  Whereas if I put the type parameter on the trait, then it forces the application developer to define their Alerts up front, when constructing the `Environment`.